### PR TITLE
Rename group

### PR DESCRIPTION
**Pull request by kba**
With `ocrd workspacec rename-group OLD NEW`, the files (if locally available) in `OLD/` are moved to `NEW/`, the resp. `mets:file` adapted and all image references in the PAGE updated accordingly.

### CODE REVIEW

**Review by bertsky — APPROVED** [review]
Excellent!

(Local paths might also become incorrect for other file formats besides PAGE, like ALTO or COCO. But we could still support this in the future. In the meantime, the user is responsible, and one can always rename back if necessary.)

BTW, Is there some use-case where we only want the METS action, but no FS and PAGE changes?

**Comment by mikegerber** [comment]
I think I found a bug using this workspace: https://qurator-data.de/examples/actevedef_718448162.first-page.zip

After:
~~~
ocrd workspace rename-group OCR-D-IMG XXX
10:58:11.967 INFO ocrd.workspace.rename_file_group - mkdir XXX
10:58:11.967 INFO ocrd.workspace.rename_file_group - Moving files
10:58:11.968 INFO ocrd.workspace.rename_file_group - Renaming file references in PAGE-XML <OcrdFile fileGrp=OCR-D-GT-PAGE, ID=OCR-D-GT-PAGE_00000024, mimetype=application/vnd.prima.page+xml, url=OCR-D-GT-PAGE/OCR-D-GT-PAGE_00000024.xml, local_filename=OCR-D-GT-PAGE/OCR-D-GT-PAGE_00000024.xml]/> 
10:58:12.180 INFO ocrd.workspace.rename_file_group - Rename pc:Page/@imageFilename: OCR-D-IMG/OCR-D-IMG_00000024.tif -> XXX/OCR-D-IMG_00000024.tif
10:58:12.182 INFO ocrd.workspace.rename_file_group - PAGE-XML changed, writing OCR-D-GT-PAGE/OCR-D-GT-PAGE_00000024.xml
10:58:12.288 INFO ocrd.workspace.rename_file_group - rmdir OCR-D-IMG
~~~

The directory and the PAGE-XML is changed:
~~~
% ls 
mets.xml  OCR-D-GT-PAGE  XXX
% grep -l XXX OCR-D-GT-PAGE/OCR-D-GT-PAGE_00000024.xml
OCR-D-GT-PAGE/OCR-D-GT-PAGE_00000024.xml
~~~

But METS is unchanged (note the date):
~~~
% ls -l mets.xml                                                                            PR-655 ?
-rw-------. 1 mike mike 21126 Jun 19  2020 mets.xml
% grep -l XXX mets.xml
%
~~~

File: ocrd/ocrd/cli/workspace.py [hunk 1]
```diff
@@ -387,6 +387,21 @@ def workspace_remove_file(ctx, id, force, keep_file):  # pylint: disable=redefin
...
+@click.argument('OLD', nargs=1)
+@click.argument('NEW', nargs=1)
+@pass_workspace
+def rename_group(ctx, old, new):
+    """
+    Rename fileGrp (USE attribute ``NEW`` to ``OLD``).
+    """
+    workspace = Workspace(ctx.resolver, directory=ctx.directory, mets_basename=basename(ctx.mets_url))
+    workspace.rename_file_group(old, new)
+
```
Comment (mikegerber) [inline ocrd/ocrd/cli/workspace.py:404]: I think there is a `workspace.save_mets()` missing here

Comment (mikegerber) [reply ocrd/ocrd/cli/workspace.py:404]: Yep, that change fixes the issue. @kba Do you add it? I think I cannot push to the PR

Comment (mikegerber) [reply ocrd/ocrd/cli/workspace.py:404]:
~~~
diff --git a/ocrd/ocrd/cli/workspace.py b/ocrd/ocrd/cli/workspace.py
index ccc199f4..a0defb4e 100644
--- a/ocrd/ocrd/cli/workspace.py
+++ b/ocrd/ocrd/cli/workspace.py
@@ -401,6 +401,7 @@ def rename_group(ctx, old, new):
     """
     workspace = Workspace(ctx.resolver, directory=ctx.directory, mets_basename=basename(ctx.mets_url))
     workspace.rename_file_group(old, new)
+    workspace.save_mets()
 
 # ----------------------------------------------------------------------
 # ocrd workspace remove-group
~~~

**Comment by mikegerber** [comment]
There is also a build fail on Python 3.5:
https://github.com/OCR-D/core/runs/1644815103

~~~
/usr/local/lib/python3.5/site-packages/ocrd/resolver.py:14: in <module>
    from ocrd.workspace import Workspace
E     File "/usr/local/lib/python3.5/site-packages/ocrd/workspace.py", line 235
E       new_url = sub(rf'^{old}/', f'{new}/', mets_file.url)
E                               ^
E   SyntaxError: invalid syntax
~~~

File: ocrd/ocrd/cli/workspace.py (hunk 1 above)
Comment (kba) [reply ocrd/ocrd/cli/workspace.py:404]: I'll add a fix, thanks for spotting this.

**Comment by kba** [comment]
> There is also a build fail on Python 3.5:

I'll fix it this time but we will stop supporting 3.5 by end of February 2021.

File: ocrd/ocrd/cli/workspace.py (hunk 1 above)
Comment (kba) [reply ocrd/ocrd/cli/workspace.py:405]: 485d284d

**Comment by mikegerber** [comment]
> > There is also a build fail on Python 3.5:
> Thanks. I'll fix it this time but we will stop supporting 3.5 by end of February 2021.

Yeah, I wasn't sure what is planned with Python 3.5.

**Pull request merged**